client/lib/explat/internals/naive-client-side-rollout.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Internal dependencies
+ */
+import user from 'calypso/lib/user';
+
+/**
+ * Naively ship a feature to a percentage of our users.
+ * - Very basic, this exists to replace the last remaining calypso-abtest use-case.
+ * - Works best for logged-in users.
+ * - Uses randomisation for logged out users.
+ * - Won't work across the log-in boundary.
+ *
+ * @param percentage 0-100
+ * @returns Boolean of whether the user should have the feature enabled, i.e. true = rolled out, false = not rolled out.
+ */
+export default function naiveClientSideRollout( percentage: number ): boolean {
```

Comment:
In case this ends up getting used elsewhere, how about we attempt to decorrelate different use cases by including a `featureName` string and hashing it along with the user ID?

Reply:
Adding an unused featureName to the function is not a bad idea just for tracking purposes. Hashing is a bit more complicated on the client side of things and isn't necessary unless we are trying to continue the allocations given to an experiment.

Reply:
:man_facepalming: I didn't realise that the legacy Calypso abtest library simply used the user ID modulo `allocationTotal`. Given how long it's been around, it's a certain recipe for carryover effects.

Given the scope of this PR, I agree that we don't need to do anything fancier.

How about we add the `featureName` as an argument and also warn in development mode that the implementation is incomplete in case someone tries to use it for anything new?